frontend/src/components/smart-contracts/modals/hooks/useReportJob.ts

```diff
@@ -0,0 +1,19 @@
+import { useSmartContracts } from 'components/smart-contracts/hooks/useSmartContracts';
+import { useJob } from '../../hooks/useJob';
+
+export const useReportJob = () => {
+  const { job } = useJob();
+
+  const { contracts } = useSmartContracts();
+
+  // the logic called to initiate the transaction
+  const callContract = async () => {
+    if (job?.id && job?.ipfsCid) {
+      return contracts.Job.reportJob(job.id, job.ipfsCid);
```

Comment:
The purpose of the ipfs cid here is to report a job with a reason metadata reference.  The flow is: upload a reason object to IPFS, get the new CID, attach that CID to the report.

This way the admin can review the report and read the reason as to why it was reported.

Reply:
Ohhh ok. I totally misunderstood that.